Adding UPS OpenShift template

secondsun opened the pull request:
@matzew As requested I have added my openshift template for great pr review

matzew commented on openshift/ups-template.json:
```diff
@@ -0,0 +1,529 @@
+  "metadata": {
+    "name": "redhat-mobile-backend",
+    "annotations": {
+      "description": "An example Mobile Backend using RedHat, JBoss, AeroGear, and FeedHenry technologies",
```
In the `description`, what is the FeedHenry part in this template?

Also, regarding the _name_, I think I'd have `name`d it like _aerogear-unified-push_

secondsun replied:
Yeah, this code is part of something else I am working on, the UPS integration was just the first piece I got done.  It just so happened I did it when a bunch of people on aerogear-dev were asking about it.

matzew replied:
ok, but still, let's change that to not confuse AG community members ;-)

josemigallas commented on openshift/ups-template.json:
```diff
@@ -0,0 +1,529 @@
+        }
+      }
+    },
+	
```
Wrong indentation here?

josemigallas commented:
@secondsun I detected wrong format along the json file so instead of pointing out every small detail I sent you a PR to your PR: https://github.com/secondsun/aerogear-unifiedpush-server/pull/1

wtrocki commented on openshift/ups-template.json:
```diff
@@ -0,0 +1,529 @@
+	    "name": "ups-localhost-unsecure"
+	  },
+	  "spec": {
+	    "host": "ups.127.0.0.1.nip.io",
```
Extract this to parameter or leave empty

wtrocki commented on openshift/ups-template.json:
```diff
@@ -0,0 +1,529 @@
+	  "status": {
+	    "ingress": [
+	      {
+		"host": "ups.127.0.0.1.nip.io",
```
when exporting template all status sections should be removed as they aren't applied to new infrastructure.

wtrocki commented on openshift/ups-template.json:
```diff
@@ -0,0 +1,529 @@
+            "containers": [
+              {
+                "name": "unifiedpush-server",
+                "image": "aerogear/unifiedpush-wildfly",
```
[Pattern] Extract all images into parameters.

wtrocki commented on openshift/ups-template.json:
```diff
@@ -0,0 +1,529 @@
+        ],
+        "resources": {
+          "requests": {
+            "storage": "1Gi"
```
Extract to template parameter.

wtrocki commented on openshift/ups-template.json:
```diff
@@ -0,0 +1,529 @@
+      },
+      "stringData": {
+        "database-user": "unifiedpush",
+        "database-password": "unifiedpush",
```
Extract to parameters.

secondsun replied:
I don't think I will be able to extract these to parameters without updating the UPS image.  Right now these values are hard coded into the liquibase migration scripts.

@matzew WDYT?

wtrocki commented on openshift/ups-template.json:
```diff
@@ -0,0 +1,529 @@
+      "kind": "DeploymentConfig",
+      "apiVersion": "v1",
+      "metadata": {
+        "name": "keycloak"
```
It's mysql for keycloak

dimitraz commented:
@secondsun Hi Summers, when I try to run the template I get this error:
```
Unexpected error running Liquibase: com.mysql.jdbc.exceptions.jdbc4.CommunicationsException: Communications link failure

The last packet sent successfully to the server was 0 milliseconds ago. The driver has not received any packets from the server.
```

Could this be due the Image Streams?

wtrocki commented on openshift/ups-template.json:
```diff
@@ -0,0 +1,529 @@
+                "mysql"
+              ],
+              "from": {
+                "kind": "ImageStreamTag",
```
Let's not use imageStreamTags and use image directly. We setup this standard in the core because multiple problems with streams. We do not benefit from updates, as they would need to be manual anyway so streams should not be used.

wtrocki commented:
In this aproach we using 2 containers for mysql - I'm not saying that this is wrong aproach but for community testing it would be better to pick less resource intensive aproach (single mysql).

wtrocki commented on openshift/ups-template.json:
```diff
@@ -0,0 +1,529 @@
+            "containers": [
+              {
+                "name": "mysql",
+                "image": " ",
```
If we remove image stream we can setup proper image here and possibly extract it as parameter.

dimitraz commented on openshift/ups-template.json:
```diff
@@ -0,0 +1,529 @@
+              "from": {
+                "kind": "ImageStreamTag",
+                "name": "mysql:5.7",
+                "namespace": "${NAMESPACE}"
```
This didn't work for because I didn't have the image already pulled. I think it's pulling it from openshift/mysql:5.7, which doesn't exist?

wtrocki replied:
This is happening because of imageStream not being available. Removing image streams will resolve problem

secondsun commented:
Quick update,

I've updated the description, merged a reformat commit, removed image streams, and renamed the keycloak mysql instance to be more explicitly keycloak_mysql.

@wtrocki I haven't externalized things to parameters yet, but that should be soon (TM)

secondsun commented:
@wtrocki I have updated the last parameters.  Let me know if it looks good.

@matzew Can you take a look and give your blessing or not.

wtrocki commented:
@secondsun I cannot verify this template at the moment. Template looks ok to me.

matzew commented:
:+1:   nice work, @secondsun @wtrocki and @dimitraz

dimitraz commented:
@secondsun @matzew Let me know if you want me to write up a quick readme or if you have that handled already 👍🏿

secondsun commented:
@dimitraz Rock the readme, I forgot about it

dimitraz commented:
@secondsun cool on it now
